effort_controllers/test/test_joint_group_effort_controller.cpp

```diff
@@ -58,116 +63,183 @@ void JointGroupEffortControllerTest::TearDown()
-
   SetUpController();
-
-  // configure failed, 'joints' paremeter not set
-  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
-  controller_->lifecycle_node_->declare_parameter(
-    "joints",
-    rclcpp::ParameterValue(std::vector<std::string>()));
+  controller_->get_node()->set_parameter({"joints", std::vector<std::string>()});
 
   // configure failed, 'joints' is empty
   ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
-  auto result = controller_->lifecycle_node_->set_parameter(
-    rclcpp::Parameter(
-      "joints",
-      rclcpp::ParameterValue(std::vector<std::string>{"joint1", "joint2"})));
-  ASSERT_TRUE(result.successful);
+}
+
+TEST_F(JointGroupEffortControllerTest, ConfigureAndActivateParamsSuccess)
+{
+  SetUpController();
+  controller_->get_node()->set_parameter({"joints", joint_names_});
 
   // configure successful
   ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+}
+
+TEST_F(JointGroupEffortControllerTest, ActivateWithWrongJointsNamesFails)
+{
+  SetUpController();
+  controller_->get_node()->set_parameter({"joints", std::vector<std::string>{"joint1", "joint4"}});
+
+  // activate failed, 'joint4' is not a valid joint name for the hardware
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
 
-  // joint handles initialized
-  ASSERT_EQ(controller_->joint_cmd_handles_.size(), 2ul);
-  ASSERT_EQ(controller_->joint_cmd_handles_[0].get_interface_name(), "effort_command");
-  ASSERT_EQ(controller_->joint_cmd_handles_[1].get_interface_name(), "effort_command");
+  controller_->get_node()->set_parameter({"joints", std::vector<std::string>{"joint1", "joint2"}});
+
+  // activate failed, 'acceleration' is not a registered interface for `joint1`
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+  ASSERT_EQ(controller_->on_activate(rclcpp_lifecycle::State()), CallbackReturn::ERROR);
 }
 
-TEST_F(JointGroupEffortControllerTest, CheckParamsTest)
+TEST_F(JointGroupEffortControllerTest, CommandSuccessTest)
 {
-  // joint handles not initialized yet
-  ASSERT_TRUE(controller_->joint_cmd_handles_.empty());
+  SetUpController();
+  controller_->get_node()->set_parameter({"joints", joint_names_});
+  ASSERT_EQ(controller_->on_configure(rclcpp_lifecycle::State()), CallbackReturn::SUCCESS);
+
+  // update successful though no command has been send yet
+  ASSERT_EQ(controller_->update(), controller_interface::return_type::SUCCESS);
 
+  // check joint commands are still the default ones
+  ASSERT_EQ(joint_1_cmd_.get_value(), 1.1);
+  ASSERT_EQ(joint_2_cmd_.get_value(), 2.1);
+  ASSERT_EQ(joint_3_cmd_.get_value(), 3.1);
```

Comment:
there is a specialized test for this and you repeat this snippet multiple times. Is that needed?
```suggestion
```